src/codegen/optimizer.lisp

```diff
@@ -21,7 +21,10 @@
   ;; Run direct application twice so that options for direct
   ;; application are not removed by the pointfree transform
   (let* ((node (direct-application-transform node optimizer))
-         (node (pointfree-transform node optimizer))
+         ;; Classes cannot be stack allocated
+         (node (if (eql coalton-impl:*interaction-mode* ':release)
```

Comment:
I'm seeing this `(if (eql coalton-impl:*interaction-mode* ':release) ...` a lot. Would it make sense to have a `with-release-mode` and `with-development-mode` or something similar?

Reply:
I don't see that as being easier to read, but maybe I'm not really understanding what you're suggesting. Would that look like this?

```lisp
(with-release-mode
  (do stuff for release mode)
  (do stuff for development mode))
```

Reply:
I myself would prefer:

```
(ecase *development-mode*
  (:development ...)
  (:release ...))
```

so that we are always explicit about what is what, and we have no other cases